tests/unit/models/gpflow/test_models.py

```diff
@@ -528,6 +546,41 @@ def test_variational_gaussian_process_update_q_mu_sqrt_unchanged() -> None:
     npt.assert_allclose(old_q_sqrt, new_q_sqrt, atol=1e-5)
 
 
+@random_seed
+def test_variational_gaussian_process_trajectory_sampler_has_correct_samples() -> None:
+    x_observed = tf.constant(np.arange(10).reshape((-1, 1)), dtype=gpflow.default_float())
+    y_observed = _3x_plus_gaussian_noise(x_observed)
+    optimizer = BatchOptimizer(tf.optimizers.Adam(), max_iter=20)
+    likelihood = gpflow.likelihoods.Gaussian(1e-5)
+    kernel = gpflow.kernels.Matern32(lengthscales=0.2)
+    vgp = VGP((x_observed, y_observed), kernel, likelihood)
+    model = VariationalGaussianProcess(vgp, optimizer=optimizer, use_natgrads=True)
+    model.update(Dataset(x_observed, y_observed))
+    model.optimize(Dataset(x_observed, y_observed))
+
+    num_samples = 100
+    trajectory_sampler = model.trajectory_sampler()
+    assert isinstance(trajectory_sampler, DecoupledTrajectorySampler)
+
+    trajectory = trajectory_sampler.get_trajectory()
+    x_predict = tf.constant([[1.0], [2.0], [3.0], [1.5], [2.5], [3.5]], gpflow.default_float())
+    x_predict_parallel = tf.expand_dims(x_predict, -2)  # [N, 1, D]
+    x_predict_parallel = tf.tile(x_predict_parallel, [1, num_samples, 1])  # [N, B, D]
+    samples = trajectory(x_predict_parallel)  # [N, B]
+    sample_mean = tf.reduce_mean(samples, axis=1, keepdims=True)
+    sample_variance = tf.math.reduce_variance(samples, axis=1, keepdims=True)
+
+    true_mean, true_variance = model.predict(x_predict)
+
+    # test predictions approx correct away from data
+    npt.assert_allclose(sample_mean[3:] + 1.0, true_mean[3:] + 1.0, rtol=0.1)
```

Comment:
whats up with + 1.0?

Reply:
It made the tests a little more stable. Better than asserting two things close to 0 are relatively close